### PR TITLE
User branch

### PR DESCRIPTION
**Pull request by ayanourr**
Overview
This PR implements the user management endpoints in our FastAPI application, creates the necessary data models, and enhances the security of the user-related functionalities.
Endpoints :
Get All Users Endpoint:
Implemented pagination for user retrieval.
Improved error handling to ensure reliable responses.

Create User Endpoint:
Added validation to prevent duplicate email registrations.
Enhanced password hashing process.

Get User Details Endpoint:
Simplified access control checks for user details based on roles.
Integrated HATEOAS links for better API navigation.

Update User Endpoint:
Improved validation for updating user information.
Updated timestamp handling to reflect modifications accurately.

Change User Role Endpoint:
Added error handling for non-existent users.
Improved logic for updating user roles.

### CODE REVIEW

**Review by FatmaQunnies24 — APPROVED** [review]

File: app/api/main.py [hunk 1]
```diff
@@ -1,9 +1,7 @@
 from fastapi import APIRouter
-
 from app.api.routes import user, login, status, order
 
 api_router = APIRouter()
 api_router.include_router(login.router, prefix="/login", tags=["login"])
 api_router.include_router(user.router, prefix="/users", tags=["users"])
-api_router.include_router(status.router, prefix="/statuses", tags=["statuses"])
```
Comment (Mohammed-samaheen) [inline app/api/main.py:8]: why did you delete the router for statuses?

File: app/api/routes/user.py [hunk 2]
```diff
@@ -1,8 +1,225 @@
...
+
+            {
+                "id": user["user_id"],
+                "username": user["username"],
+                "email": user["email"],
+                "is_admin": user["is_admin"],
+                "is_active": user["is_active"],
+                "created_at": user["created_at"],
+                "updated_at": user["updated_at"],
+                "links": GetUserDetailsResponse.create_hateoas_links(user["user_id"])
```
Comment (Mohammed-samaheen) [inline app/api/routes/user.py:65]: why did you add links?

Comment (ayanourr) [reply app/api/routes/user.py:65]: I added HATEOAS links to make the API more flexible . This lets the frontend follow the links provided instead of hardcoding URLs, so if we change any routes in the backend, the frontend won’t break. It also helps the API show what actions are available for each resource.

File: app/api/main.py (hunk 1 above)
Comment (ayanourr) [reply app/api/main.py:8]: I deleted it by mistake, I'll add it back

**Review by Mohammed-samaheen — APPROVED** [review]
